splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/DropTriggerConstantOperation.java

```diff
@@ -35,124 +35,126 @@
+                                tableId.toString());
+        }
+
+        /*
+        ** Get the trigger descriptor.  We're responsible for raising
+        ** the error if it isn't found
+        */
+        triggerd = dd.getTriggerDescriptor(triggerName, sd);
+
+        if (triggerd == null) {
+            throw StandardException.newException(SQLState.LANG_OBJECT_NOT_FOUND_DURING_EXECUTION, "TRIGGER",
+                    (sd.getSchemaName() + "." + triggerName));
+        }
+
+        /*
+         ** Prepare all dependents to invalidate.  (This is there chance
+        ** to say that they can't be invalidated.  For example, an open
+        ** cursor referencing a table/trigger that the user is attempting to
+        ** drop.) If no one objects, then invalidate any dependent objects.
+        */
 
         DependencyManager dm = dd.getDependencyManager();
         dm.invalidateFor(triggerd, DependencyManager.DROP_TRIGGER, lcc);
 
         // Drop the spses
-        SPSDescriptor spsd = dd.getSPSDescriptor(triggerd.getActionId());
-        dm.invalidateFor(spsd, DependencyManager.DROP_TRIGGER, lcc);
+        for (UUID actionId: triggerd.getActionIdList()) {
+            SPSDescriptor spsd = dd.getSPSDescriptor(actionId);
+            dm.invalidateFor(spsd, DependencyManager.DROP_TRIGGER, lcc);
 
-        DDLMessage.DDLChange ddlChange = ProtoUtil.dropTrigger(((SpliceTransactionManager) tc).getActiveStateTxn().getTxnId(),
-                (BasicUUID) this.tableId, (BasicUUID) triggerd.getUUID(),
-                (BasicUUID) triggerd.getActionId());
-        // Run Remotely
-        tc.prepareDataDictionaryChange(DDLUtils.notifyMetadataChange(ddlChange));
+            DDLMessage.DDLChange ddlChange = ProtoUtil.dropTrigger(((SpliceTransactionManager) tc).getActiveStateTxn().getTxnId(),
+                    (BasicUUID) this.tableId, (BasicUUID) triggerd.getUUID(),
+                    (BasicUUID) actionId);
+            // Run Remotely
+            tc.prepareDataDictionaryChange(DDLUtils.notifyMetadataChange(ddlChange));
 
-        dm.clearDependencies(lcc, triggerd);
-        dm.clearDependencies(lcc, spsd);
+            dm.clearDependencies(lcc, triggerd); // XXX arnaud get some stuff out of the loop
+            dm.clearDependencies(lcc, spsd);
 
-        // Drop the trigger
-        dd.dropTriggerDescriptor(triggerd, tc);
+            // Drop the trigger
+            dd.dropTriggerDescriptor(triggerd, tc);
 
-        // there shouldn't be any dependencies, but in case there are, lets clear them
-        dd.dropSPSDescriptor(spsd, tc);
-        // Remove all TECs from trigger stack. They will need to be rebuilt.
-        lcc.popAllTriggerExecutionContexts();
+            // there shouldn't be any dependencies, but in case there are, lets clear them
+            dd.dropSPSDescriptor(spsd, tc);
+            // Remove all TECs from trigger stack. They will need to be rebuilt.
```

Comment:
Should we move these statements out of the loop?

Reply:
Done